sdk/identity/test-resources-pre.ps1

```diff
@@ -0,0 +1,39 @@
+[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    # Captures any arguments from eng/New-TestResources.ps1 not declared here (no parameter errors).
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments,
+
+    [Parameter()]
+    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+    [string] $TestApplicationId,
+
+    [Parameter()]
+    [string] $TestApplicationSecret,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $TenantId
+)
+
+pip install azure-cli=="2.56.0" | Write-Host
```

Comment:
I'm surprised this isn't a pipeline step or something that is in the ambient state of our VMs? Are we just pinning a version to work around a bug? Maybe a comment here at least to explain why it is here.

Reply:
added the comment.